### ISSUE TITLE
Reactome Release 82

### ISSUE THREAD

**Issue by ukemi** [geneontology/pathways2GO#176]
- [x] 1. ONGOING- Tickets on GOC side to improve import process
                    - [ ] - Replace transports_or_maintains_localization_of relations with has_primary_input #200. I believe we also still need to update the Shex. https://github.com/geneontology/go-shapes/issues/273
                    - [X] - https://github.com/geneontology/pathways2GO/issues/173
                    - [ ] - Reexamine the filtering for drugs: revisit our curation practice and the GO-CAM conversion process. I suspect that we want to be able to assign broader BP terms to pathways and more specific BP terms to reactions contained in those pathways, but we don't always want to do this, so we will need to discuss with Dustin how to do this without confusing the GO-CAM conversion process. R-HSA-974878, R-HSA-5620971,  Still say 'NO' to drugs #182 
                    - [X] - https://github.com/geneontology/go-site/pull/1902
                    - [ ] - generate stats for fraction of GO-CAMs that is complete and the reasons for incompleteness (like the Venn diagram Figure 4(?) in the Good et al. paper)
- [x] 2. 15/09/2022- BioPax available for test load into Noctua-dev
- [x] 3. 11/10/2022- Load Biopax into noctua-dev
- [x] 4. 11/10/2022- Run Shex QC and Logical error QC on GO-CAMS in noctua-dev.
- [x] 5. DATE- Fix logical and QC errors at Reactome. 
- [x] 6. 2022-07(18-29)- merge final changes that have been added to the GOC conversion/import code. Can occur after 8.
- [x] 7. 2022-07-22- Regenerate BioPax--- won't happen this round
- [x] 8. 2022-07-(25-29)- Rerun BioPax load into Noctua dev---won't happen this round
- [x] 9. 2022-08-01- Rerun Shex and logical checks- should be clean. If not, return to 4 or punt to next release.-- won't happen this round
- [x] 10. 2022-08-03- Fix any issues with the GOC conversion code and rerun load and checks. Must be done before 13.
- [x] 11. 2022-08-05 - Reactome Data freeze - target date to complete 1 - 9?
- [x] 12. 2022-08-24 - Reactome final slice - absolute deadline for completing 1 - 9.
- [x] 13. 2022-09-14 - Reactome Release
- [x] 14. 2022-09-22- Load Reactome Release BioPax into Noctua prod
- [x] 15. 2022-09-22/3- GO-CAM model release
- [x] 16. 2022-09-23/4 - spot-check released models for correctness

**Comment by ukemi** [comment]
@deustp01 and @dustine32 we need to add dates to the task list above. All of the things under 5 should get turned into tickets. If we can do incremental testing, we can check them off as we go.

**Comment by dustine32** [comment]
Do we have the release 82 BioPAX Homo_sapiens.owl file available now?

**Comment by deustp01** [comment]
> Do we have the release 82 BioPAX Homo_sapiens.owl file available now?

Not yet. Should be available next week, maybe as early as 9/12

**Comment by dustine32** [comment]
@deustp01 Thanks! I can just wait until that's available and then push the new models to noctua-dev for testing.

**Comment by ukemi** [comment]
Thanks guys. So Wednesday we will need to talk about a modification of our SOP for the new loads since the BioPax isn't available until the release.

**Comment by deustp01** [comment]
> new loads since the BioPax isn't available until the release.

In the future, maybe as soon as the next release, we should have the BioPAX in advance, as planned, so I hope we are looking aty a delay in our plans, not a change.

**Comment by dustine32** [comment]
@ukemi @deustp01 ShEx and OWL consistency checks have been run. 5 models failed ShEx, all are logically consistent. Full `main_report.txt` is below along with `explanations.txt` for the failures:
[main_report.txt](https://github.com/geneontology/pathways2GO/files/9759716/main_report.txt)
[explanations.txt](https://github.com/geneontology/pathways2GO/files/9759718/explanations.txt)

You can find the five ShEx failing models in `main_report.txt` by sorting the `shex_valid` column. Here are model links for your convenience:
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-9670095
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-9708296
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-1474151
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-4615885
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-2514859

**Comment by ukemi** [comment]
I've gone through these and here is what I think. I should probably still tweak the models and not save to see if I am right. @dustine32 There are a couple cases below where I don't see why some Reacto entities aren't passing.

http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-9670095
Annotation to an obsolete term. This should be corrected?
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-9708296
Endoribonuclease activity input and output of REACTO_R-HSA-9708818 and REACTO_R-HSA-9708815 are not typed as chemical entities or complexes. Why?
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-1474151
oxidoreductase activity has input REACTO_R-HSA-9693721. This should be CHEBI:17804. Why isn't it?
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-4615885
RANBP2 SUMOylates CDCA8 (Borealin) and PIAS3 SUMOylates AURKB (Aurora-B). This is a black box reaction with two catalysts. I suspect this is what is causing the Shex failure. Both are mapped to SUMO transferase activity. MFs should only have one enabler.
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-2514859
glycine N-acyltransferase activity has an input of obo:go/shapes/ProteinContainingComplex, obo:go/shapes/InformationBiomacromolecule, but this reaction also has an input of Acyl-CoA. @vanaukenk, we might want to modify the Shex here and make the target proteins a primary input. These kinds of enzymes will have molecules other than the target protein/complex being modified.

**Comment by dustine32** [comment]
Thank you @ukemi for testing and the feedback! Sorry for the delay in responding. This was quite the learning experience and my responses below went through several iterations of "ohhhh!" and then having to rewrite what's going on.

So, for our failing models:
* http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-9670095
  * Right, the term _should_ be replaced but [GO:1900051](https://www.ebi.ac.uk/QuickGO/term/GO:1900051) doesn't have the `term replaced by` (`IAO:0100001`) annotation in the GO ontology, which is currently required by the conversion code. There two "consider" terms. To fix, should the conversion code be looking at these "consider" annotations or should the term get updated in the GO with the usual `term replaced by`?
* http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-9708296
  * The input [R-HSA-9708815](https://reactome.org/content/detail/R-HSA-9708815) and output [R-HSA-9708818](https://reactome.org/content/detail/R-HSA-9708818) are both sets. There may be something missing in getting ShEx to recognize these. In `reacto.owl`, these classes do not have any `subClassOf` relationship to CHEBI or GO terms that ShEx could use to infer they are either a `ChemicalEntity` or `ProteinContainingComplex`. I wonder if there are other models that use sets w/o issue.
* http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-1474151
  * Yes, I see in the BioPAX that input [PTHP](https://reactome.org/content/detail/R-ALL-1474179#Homo%20sapiens) has `CHEBI:17804` xref'd to its `entityReference` (in the External Reference Information section). However, the conversion code currently doesn't look at xrefs from `entityReference` elements on a SmallMolecule object and instead just uses its Reactome ID. Same with output [sepiapterin](https://reactome.org/content/detail/R-ALL-1497811#Homo%20sapiens) and likely every other small molecule in Reactome GO-CAMs. We can open a ticket to change this behavior to always fetch the CHEBI if that is desired.
  * For the enabled_by (I think this is the real ShEx violation), sepiapterin synthase (R-HSA-9693721) is in the BioPAX as a PhysicalEntity. See its [entry](https://reactome.org/content/detail/R-HSA-9693721) at Reactome and notice it does not have a CHEBI cross reference. As a result, in `reacto.owl`, R-HSA-9693721 only has `subClassOf` continuant, which is not specific enough to be inferred as either InformationBiomacromolecule or ProteinContainingComplex.
* http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-4615885
  * Should this single activity be broken into two activities to accommodate both enablers? Or just keep the single activity and select the best enabler, however "best" is defined?
* http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-2514859
  * enabler "unknown NAT" [R-HSA-2565924](https://reactome.org/content/detail/R-HSA-2565924) is also missing a CHEBI cross ref in Reactome, causing it to only get a `subClassOf continuant` in `reacto.owl`.

**Comment by ukemi** [comment]
@dustine32 I just met with @deustp01 and he reminded me of some things we had done with other models. He will comment here, but in the meantime don't work on my suggestion to use ChEBI just yet.

**Comment by ukemi** [comment]
@dustine32 and @deustp01 Having a night to think this over, I think we should adopt a strategy to use the ChEBI identifiers. Peter, after our conversation yesterday the realization struck me that when I use the model copy functionality all of the Reactome chemicals get copied over to my models. See http://noctua.geneontology.org/editor/graph/gomodel:633b013300001469?
This means that I need to go in and clean up all my models where I've used model copy. Less than an ideal situation.

**Comment by deustp01** [comment]
@dustine32 and @ukemi Still thinking here about the correct role for ChEBI identifiers in these models - needs more discussion.

On the five models that failed ShEX and OWL, it looks like all have straightforward, known errors in the Reactome annotation / BioPAX export that are easy enough to fix on the Reactome side so that, if such turned up at QA time in a future fast Reactome-to-GO-CAM export, we could make the fixes within the seven-day window that should be available. Here, I have fixed four already; the fifth requires strong-arming a curator. Detailed comments for each are interpolated into Dustin's comment on 10/14.

**Comment by dustine32** [comment]
OK, great! Thanks @ukemi for summing up the actions. If we decide to switch to extracting CHEBI IDs we can work off of that new code change ticket.

**Comment by ukemi** [comment]
@deustp01, should we move forward? @dustine32, during a weeds call last week, we decided that we should go ahead and try to import the ChEBI identifiers that are xref'd in Reactome. The main reason for this is consistency. Right now if I copy a model to make the mouse version, the Reactome entities come into my model. If I then make the model production, the Reactome entities end up on the Alliance view. This wouldn't be terrible if we could link them back to Reactome from there, but the group decided that it would be better if they linked to ChEBI because the ChEBI entities are the ones that are used by curators when they make de novo models. I will open a new ticket for this task. Meanwhile, I think the next thing to do is to sanity check the models that are on dev? and determine whether or not we can push the new load to production. Is everyone good with that?

**Comment by deustp01** [comment]
> Is everyone good with that?

I'm good with both (replace Reactome IDs for localized small molecules with ChEBI ID coupled if possible to GO CC term) and doing sanity checks on latest models before push to production.

**Comment by dustine32** [comment]
@ukemi Yep, good with both as well. Let me know when/if you are good with the models in noctua-dev and I can make a PR with the new models to `noctua-models/master`. This PR will then get merged/loaded to Noctua prod during the next Noctua maintenance outage (most likely 2022-11-10).

**Comment by ukemi** [comment]
Models to check:

- [ ] R-HSA-5661270  - A nice linear pathway
- [ ] R-HSA-70171  - Because it's my favorite
- [ ] R-HSA-75893 - An update TNF signaling pathway
- [ ] R-HSA-112311 - Search for contraband chemicals

**Comment by ukemi** [comment]
Notes for future work in release 83:

1. For links between pathways, it would be nice to infer a more specific relation than causally upstream of. In most cases, these are functions that are immediately upstream of and maybe even a directly-provides_input_for as long as that is valid for functions between processes. Something to discuss.
2. We should implement the has_small_molecule_regulator for the next release glycolysis is a good pathway to look at.
3. Don't forget that we want to convert chemicals to ChEBI
4. TNF is an unsatisfying pathway because it is all molecular events. However, we did successfully filter the drugs away. Is there any way we can resolve the Greek letters? @deustp01 I notice that sometimes they are spelled out and sometimes they are Greek. Is there a rule for Reactome curation?

**Comment by ukemi** [comment]
OK @dustine32, I checked the above four models as well as the ones that failed the Shex and didn't spot any glaring issues with the import. I made a couple of notes above just so that I can remember them. There are now only two tasks left in this ticket, both in your court:

1. Go ahead with the release to production.
2. Generate the stats for @deustp01. I think he needs them for accounting purposes. Much like pathway boundaries, I have included that in this ticket, but clearly the release can technically go through without it. So if it is a thorny task, we can go ahead and call this release done and move that ticket, but I think it should still be a priority to get those stats. @deustp01 please feel free to comment.

**Comment by dustine32** [comment]
@ukemi Great! I just found code specifically for the Venn diagram numbers in [garage/Manuscript.java](https://github.com/geneontology/pathways2GO/blob/master/exchange/src/main/java/org/geneontology/garage/Manuscript.java) that I can revive.

**Comment by ukemi** [comment]
@dustine32 I'm wondering where we stand on this release since @deustp01 might want to report on the status at tomorrow's PI meeting.

**Comment by dustine32** [comment]
@ukemi @deustp01 I figured out the dang stats!!

First, here's the Venn diagram based on the latest GO-CAM load of Reactome 82 going into Noctua prod today:
![image](https://user-images.githubusercontent.com/2678599/201226019-de006f0e-bf43-4000-85c0-964b56d8ea5a.png)
Note this Venn diagram is created by a free web tool at https://bioinformatics.psb.ugent.be/webtools/Venn/.

The commands to generate the input for this diagram tool are now incorporated into the Reactome -> GO-CAM conversion pipeline (along with the ShEx checks).

**Comment by deustp01** [comment]
Great! I will align with the published diagram.

**Comment by ukemi** [comment]
Release 82 has been moved to the 'Done" column. Onward to 83.
